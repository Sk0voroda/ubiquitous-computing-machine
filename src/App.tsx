import { lazy, Suspense } from 'react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';

import { RootLayout } from './views/common/RootLayout';
import { LoginSkeleton } from './views/LoginView';

import { Routes } from './routes';

const LoginView = lazy(() => import('@views/LoginView'));
const MainView = lazy(() => import('@views/MainView'));
const LogoutView = lazy(() => import('@views/LogoutView'));

// TODO: add text color main secondary base etc
// TODO: add pre comit hooks
const router = createBrowserRouter([
  {
    path: Routes.Main,
    element: <RootLayout />,
    children: [
      {
        path: Routes.Main,
        element: (
          <Suspense fallback={'loading'}>
            <MainView />
          </Suspense>
        ),
      },
      {
        path: Routes.Login,
        element: (
          <Suspense fallback={<LoginSkeleton />}>
            <LoginView />
          </Suspense>
        ),
      },
      {
        path: Routes.Logout,
        element: (
          <Suspense fallback="...loading">
            <LogoutView />
          </Suspense>
        ),
      },
    ],
  },
]);

export const App = () => <RouterProvider router={router} />;
