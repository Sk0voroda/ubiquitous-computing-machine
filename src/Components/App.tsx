import { useState } from "react";
import z from "zod";
import clsx from "clsx";

import { mockItems } from "../mocks";

const TODOS = "TODOS";

const ItemSchema = z.object({
  id: z.string(),
  task: z.string(),
  done: z.boolean(),
});
const ItemsSchema = z.array(ItemSchema);
type Item = z.infer<typeof ItemSchema>;

const ToDoItem: React.FC<Item> = ({ id, task, done }) => (
  <div
    id={id}
    className={`flex cursor-pointer items-center rounded-md p-2 hover:bg-blue-200 children:cursor-pointer ${clsx(
      {
        "line-through opacity-50": done,
      }
    )}`}
  >
    <input
      id={`todo-item-${id}`}
      type="checkbox"
      value=""
      className="h-4 w-4 rounded border-gray-300 bg-gray-100 text-blue-600 focus:ring-2  dark:border-gray-500 dark:bg-gray-600 dark:ring-offset-gray-700 dark:focus:ring-blue-600"
    />
    <label htmlFor={`todo-item-${id}`} className="ml-2 w-full ">
      {task}
    </label>
  </div>
);

function App() {
  // TODO: try to add dark theme later
  // const [darkMode, setDarkMode] = useState(true);
  const [items, setItems] = useState<Item[]>(() => {
    if (mockItems && mockItems.length > 0 && import.meta.env.DEV)
      return mockItems;
    if (Object.prototype.hasOwnProperty.call(localStorage, TODOS)) {
      const parsedTodos = ItemsSchema.safeParse(
        JSON.parse(localStorage.getItem(TODOS) as string)
      );

      if (parsedTodos.success) {
        return parsedTodos.data;
      }
    }
    return [];
  });

  return (
    <div className="h-screen bg-blue-100">
      <div className="container mx-auto px-2">
        <header>DoMe App</header>
        <div className="py-3 children:mb-1 last:children:mb-0">
          {items.map((item) => (
            <ToDoItem key={item.id} {...item} />
          ))}
        </div>
      </div>
    </div>
  );
}

export default App;
