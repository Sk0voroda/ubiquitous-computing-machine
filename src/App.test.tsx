import { render, screen } from '@testing-library/react';

import { App } from './App';

describe('App', () => {
  beforeEach(() => {
    render(<App />);
  });

  it('renders headline', () => {});
});
