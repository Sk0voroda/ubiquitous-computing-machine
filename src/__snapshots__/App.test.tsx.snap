// Vitest Snapshot v1

exports[`App > renders 1`] = `
<DocumentFragment>
  <header
    class="fixed top-0 left-0 right-0 z-10 overflow-hidden bg-white font-mono text-blue-600 shadow-md"
  >
    <div
      class="container mx-auto flex flex-wrap items-center justify-between py-1 px-3"
    >
      <a
        class="focus-outline flex items-center p-1 font-medium lg:items-center lg:justify-center"
        href="/"
      >
        <img
          class="h-8 w-8"
          src="/src/assets/images/the-brain-logo.svg"
        />
        <span
          class="invisible ml-0 w-0 text-xl sm:visible sm:ml-3 sm:w-auto"
        >
          WONTDO
        </span>
      </a>
      <p
        class="text-sm font-thin"
      >
        <span
          class="invisible mr-0 text-xs text-black sm:visible sm:mr-2"
        >
          Don't have an account?
        </span>
        <a
          class="focus-outline p-1 underline decoration-1 underline-offset-2 hover:no-underline hover:decoration-0"
          href="/registration"
        >
          Sign up
        </a>
      </p>
    </div>
  </header>
  <main
    class="flex h-screen overflow-auto bg-stone-100"
  />
</DocumentFragment>
`;
